### PR TITLE
chore(core): enforce use of next-intl's navigation APIs

### PR DESCRIPTION
**Pull request by jorgemoya**
## What/Why?
Noticed we were not using next-intl's wrapper navigation APIs.

- Added eslint rule to enforce use of these APIs.
- Fixed issues.

## Testing
Locally, behaves the same.

### CODE REVIEW

**Comment by jorgemoya** [comment]
Investigating how to prevent `redirect` to append locale on default locale.

**Comment by jorgemoya** [comment]
Opened an issue for this: https://github.com/amannn/next-intl/issues/1335.

Will need to expect locale prefix when using `redirect` until a solution is found.

File: core/.eslintrc.cjs [hunk 1]
```diff
@@ -27,6 +27,21 @@ const config = {
...
           },
+          {
+            name: '~/i18n/routing',
+            importNames: ['Link'],
+            message: "Please import 'Link' from '~/components/Link' instead.",
+          },
+          {
+            name: 'next/router',
+            importNames: ['useRouter'],
+            message: 'Please import from `@/i18n/routing` instead.',
```
Comment (deini) [inline core/.eslintrc.cjs:38]: Should these be `~/` instead of `@/`?

Comment (jorgemoya) [reply core/.eslintrc.cjs:38]: Yes, will update.

**Review by deini — APPROVED** [review]

**Pull request merged**